client/src/pages/SearchMovie/index.js

```diff
@@ -9,20 +9,27 @@ import { authFetch } from '../../auth'
 import Results from './Results'
 import { useMovies } from '../../providers/Movies'
 
+function checkAddedMovies(movieIds, movieData) {
+  movieData.forEach((movie) => {
```

Comment:
forEach mutates the `movieData`.
The idea behind my suggestion is that we return a new data structure without mutating movieData. 

```js
function checkAddedMovies(movieIds, movieData) {
  const transformed = movieData.map((movie) => {
    const isAdded = movieIds.includes(movie.imdbId)

    return {
      ...movie,
      isAdded,
    }
  })

  return transformed
}

```